005-debounce-decorator.js

```diff
@@ -0,0 +1,17 @@
+function debounce(fn, timeout) {
+    const timer;
+  
+    return function() {
```

Comment:
please use atomic commit approach: 1 entity one commit (in your pull request please use commit messages like 1) feat(005) add debounce decorator) and 2) feat(006) add throttle decorator)